### PR TITLE
Add a Maven profile to disable User and Groups management tab from the GUI

### PR DESCRIPTION
**Pull request by Damianofds**
This pull request Closes issue #41

### CODE REVIEW

File: src/gui/core/plugin/userui/pom.xml [hunk 1]
```diff
@@ -132,7 +132,15 @@
 				</exclusion>
 			</exclusions>
 		</dependency>
+        
+		<dependency>
+			<groupId>org.slf4j</groupId>
+			<artifactId>slf4j-log4j12</artifactId>
+			<version>1.7.7</version>
+			<scope>test</scope>
+		</dependency>
```
Comment (mbarto) [inline src/gui/core/plugin/userui/pom.xml]: why do we need slfj-lof4j12 in test scope?

File: src/gui/core/plugin/userui/pom.xml [hunk 2]
```diff
@@ -264,4 +272,33 @@
...
+								</goals>
+							</execution>
+						</executions>
+					</plugin>
+				</plugins>
+			</build>
+		</profile>
+	</profiles>
+	
 </project>
```
Comment (mbarto) [inline src/gui/core/plugin/userui/pom.xml]: Can we have better formatting (4 spaces are enough)?

File: src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/InstancesManagerServiceImpl.java [hunk 3]
```diff
@@ -31,6 +21,13 @@
...
 import org.apache.commons.httpclient.methods.GetMethod;
 import org.apache.commons.io.IOUtils;
+import org.geoserver.geofence.gui.client.ApplicationException;
+import org.geoserver.geofence.gui.client.model.GSInstance;
+import org.geoserver.geofence.gui.client.model.data.rpc.RpcPageLoadResult;
+import org.geoserver.geofence.gui.server.service.IInstancesManagerService;
+import org.geoserver.geofence.gui.service.GeofenceRemoteService;
+import org.geoserver.geofence.services.dto.ShortInstance;
+import org.geoserver.geofence.services.exception.NotFoundServiceEx;
 import org.slf4j.Logger;
```
Comment (mbarto) [inline src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/InstancesManagerServiceImpl.java]: Please avoid committing packages cleanup if this is the only change inside a class

File: src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/ProfilesManagerServiceImpl.java [hunk 4]
```diff
@@ -22,6 +13,14 @@
...
 import java.util.TreeSet;
 
+import org.geoserver.geofence.gui.client.ApplicationException;
+import org.geoserver.geofence.gui.client.model.UserGroup;
+import org.geoserver.geofence.gui.client.model.data.ProfileCustomProps;
+import org.geoserver.geofence.gui.client.model.data.rpc.RpcPageLoadResult;
+import org.geoserver.geofence.gui.server.service.IProfilesManagerService;
+import org.geoserver.geofence.gui.service.GeofenceRemoteService;
+import org.geoserver.geofence.services.dto.ShortGroup;
+import org.geoserver.geofence.services.exception.NotFoundServiceEx;
```
Comment (mbarto) [inline src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/ProfilesManagerServiceImpl.java]: Please avoid committing packages cleanup if this is the only change inside a class

File: src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/RulesManagerServiceImpl.java [hunk 5]
```diff
@@ -53,11 +58,6 @@
 import com.vividsolutions.jts.geom.Polygon;
 import com.vividsolutions.jts.io.ParseException;
 import com.vividsolutions.jts.io.WKTReader;
-import org.geoserver.geofence.core.model.IPAddressRange;
-import org.geoserver.geofence.core.model.enums.CatalogMode;
-import org.geoserver.geofence.gui.client.model.data.ClientCatalogMode;
-import org.geoserver.geofence.services.util.IPUtils;
-import org.springframework.dao.DuplicateKeyException;
```
Comment (mbarto) [inline src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/RulesManagerServiceImpl.java]: Please avoid committing packages cleanup if this is the only change inside a class

File: src/gui/core/plugin/userui/src/test/java/org/geoserver/geofence/gui/PropertiesLoading.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ */
+public class PropertiesLoading extends Assert{
+
+    @Test
+    public void testActivateUserGroup(){
+        IGsUsersManagerService service = new GsUsersManagerServiceImpl();
+        boolean outcome = service.activateUserGroupTabs();
+        assertEquals(true, outcome);
+    }
```
Comment (mbarto) [inline src/gui/core/plugin/userui/src/test/java/org/geoserver/geofence/gui/PropertiesLoading.java]: Not so meaningful as a test. Can't we think of a better test? Or maybe, suite of tests?

**Comment by mbarto** [comment]
Looks good to me, now.

File: src/gui/core/plugin/userui/pom.xml (hunk 1 above)
Comment (Damianofds) [inline src/gui/core/plugin/userui/pom.xml]: no, it was just for the not meanigfull test discussed in this pul req, I'll remove it

File: src/gui/core/plugin/userui/pom.xml (hunk 2 above)
Comment (Damianofds) [inline src/gui/core/plugin/userui/pom.xml]: well, the formatting looks weird at a first glance but I have double checked and it use the same indentation of the rest of this file...

File: src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/InstancesManagerServiceImpl.java (hunk 3 above)
Comment (Damianofds) [inline src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/InstancesManagerServiceImpl.java]: ok

File: src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/ProfilesManagerServiceImpl.java (hunk 4 above)
Comment (Damianofds) [inline src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/ProfilesManagerServiceImpl.java]: ok

File: src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/RulesManagerServiceImpl.java (hunk 5 above)
Comment (Damianofds) [inline src/gui/core/plugin/userui/src/main/java/org/geoserver/geofence/gui/server/service/impl/RulesManagerServiceImpl.java]: ok

File: src/gui/core/plugin/userui/src/test/java/org/geoserver/geofence/gui/PropertiesLoading.java (hunk 6 above)
Comment (Damianofds) [inline src/gui/core/plugin/userui/src/test/java/org/geoserver/geofence/gui/PropertiesLoading.java]: removed

**Comment by Damianofds** [comment]
see this revisioned pull request: https://github.com/geoserver/geofence/pull/44 I'm close this pull req now...

**Pull request closed without merging**